keps/sig-network/0013-20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -0,0 +1,626 @@
+            * [Multiple cluster CIDRs configuration](#multiple-cluster-cidrs-configuration)
+      * [IPVS Support and Operation](#ipvs-support-and-operation)
+      * [Health/Liveness/Readiness Probes for Dual-Stack Pods](#healthlivenessreadiness-probes-for-dual-stack-pods)
+      * [CoreDNS Operation](#coredns-operation)
+      * [Ingress Controller Operation](#ingress-controller-operation)
+         * [GCE Ingress Controller: Out-of-Scope, Testing Deferred For Now](#gce-ingress-controller-out-of-scope-testing-deferred-for-now)
+         * [NGINX Ingress Controller - Dual-Stack Support for Bare Metal Clusters](#nginx-ingress-controller---dual-stack-support-for-bare-metal-clusters)
+      * [Load Balancer Operation](#load-balancer-operation)
+         * [Type ClusterIP](#type-clusterip)
+         * [Type NodePort](#type-nodeport)
+         * [Type Load Balancer](#type-load-balancer)
+      * [Cloud Provider Plugins Considerations](#cloud-provider-plugins-considerations)
+         * [Multiple bind addresses configuration](#multiple-bind-addresses-configuration-1)
+         * [Multiple cluster CIDRs configuration](#multiple-cluster-cidrs-configuration-1)
+      * [Container Environment Variables](#container-environment-variables)
+      * [Kubeadm Support](#kubeadm-support)
+         * [Kubeadm Configuration Options](#kubeadm-configuration-options)
+         * [Kubeadm-Generated Manifests](#kubeadm-generated-manifests)
+      * [vendor/github.com/spf13/pflag](#vendorgithubcomspf13pflag)
+      * [End-to-End Test Support](#end-to-end-test-support)
+      * [User Stories](#user-stories)
+      * [Risks and Mitigations](#risks-and-mitigations)
+   * [Graduation Criteria](#graduation-criteria)
+   * [Implementation History](#implementation-history)
+   * [Alternatives](#alternatives)
+      * [Dual Stack at the Edge](#dual-stack-at-the-edge)
+      * [Variation: Dual-Stack Service CIDRs (a.k.a. Full Dual Stack)](#variation-dual-stack-service-cidrs-aka-full-dual-stack)
+         * [Benefits](#benefits)
+         * [Changes Required](#changes-required)
+
+## Summary
+
+This proposal adds IPv4/IPv6 dual stack functionality to Kubernetes clusters. This includes the following concepts:
+- Awareness of multiple IPv4/IPv6 address assignments per pod
+- Native IPv4-to-IPv4 in parallel with IPv6-to-IPv6 communications to, from, and within a cluster
+
+## Motivation
+
+The adoption of IPv6 has increased in recent years, and customers are requesting IPv6 support in Kubernetes clusters. To this end, the support of IPv6-only clusters was added as an alpha feature in Kubernetes Version 1.9. Clusters can now be run in either IPv4-only, IPv6-only, or in a "single-pod-IP-aware" dual-stack configuration. This "single-pod-IP-aware" dual-stack support is limited by the following restrictions:
+- Some CNI network plugins are capable of assigning dual-stack addresses on a pod, but Kubernetes is aware of only one address per pod.
```

Comment:
Thanks for the detailed design! We are running some prototype dual stack configurations inside of GCE and starting to find ways to work around the fact that Kubernetes itself is unaware of the IPv6 addresses.
I'd like very much to stay close to the design in this doc, and reinforce our prototyping/testing efforts when the time comes. Please keep me in the loop :)